GPU compilation fails with build_type=Debug

**Describe the issue**
Building CoreNEURON with GPU support enabled and the Debug CMake build type ("`-O0 -g`") produces link errors:
```
     862     => Binary creating x86_64/special-core
     863    /gpfs/bbp.cscs.ch/ssd/gitlab_map_jobs/bbpcihpcproj12/P2923/10480/sp
            ack-build/spack-stage-coreneuron-develop-gankivynkc5kvmdtjckkylv3fw
            jzizuk/spack-build-gankivy/share/coreneuron/coreneuron.cpp:
     864    @E@nvlink error   : Undefined reference to '_ZN54_INTERNAL_32_x86_6
            4_corenrn_mod2c_exp2syn_cpp_25c7f1d810coreneuron19_net_receive_kern
            elEdPNS0_13Point_processEid' in 'x86_64/libcorenrnmech.a:exp2syn.o'
     865    @E@nvlink error   : Undefined reference to '_ZN53_INTERNAL_31_x86_6
            4_corenrn_mod2c_expsyn_cpp_0127273610coreneuron19_net_receive_kerne
            lEdPNS0_13Point_processEid' in 'x86_64/libcorenrnmech.a:expsyn.o'
     866    pgacclnk: child process exit status 2: /gpfs/bbp.cscs.ch/ssd/apps/h
            pc/jenkins/deploy/externals/2021-01-06/linux-rhel7-x86_64/gcc-9.3.0
            /nvhpc-21.2-67d2qp/Linux_x86_64/21.2/compilers/bin/tools/nvdd
  >> 867    make[3]: *** [x86_64/special-core] Error 2
```
we currently use `RelWithDebInfo` instead for these builds, which works.

**To Reproduce**
Steps to reproduce the behavior:
```bash
spack install coreneuron@develop%nvhpc+gpu~report build_type=Debug
```
(`~report` is just to avoid the annoyance in the BB5 setup of trying to compile `hdf5%nvhpc`...)

**Expected behavior**
The build should succeed.

**System (please complete the following information)**
 - OS: BB5
 - Compiler: NVHPC 21.2
 - Version: master
 - Backend: GPU

--- a/CMake/MakefileBuildOptions.cmake
+++ b/CMake/MakefileBuildOptions.cmake
@@ -72,7 +72,9 @@ endforeach()
 # PGI compiler adds --c++14;-A option for C++14, remove ";"
 string(REPLACE ";" " " CXX14_STD_FLAGS "${CMAKE_CXX14_STANDARD_COMPILE_OPTION}")
 string(TOUPPER "${CMAKE_BUILD_TYPE}" _BUILD_TYPE)
-set(CORENRN_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${CMAKE_CXX_FLAGS_${_BUILD_TYPE}} ${CXX14_STD_FLAGS} ${NVHPC_CXX_INLINE_FLAGS}")
+set(CORENRN_CXX_FLAGS
+    "${CMAKE_CXX_FLAGS} ${CMAKE_CXX_FLAGS_${_BUILD_TYPE}} ${CXX14_STD_FLAGS} ${NVHPC_CXX_INLINE_FLAGS}"
+)
 
 # =============================================================================
 # nmodl/mod2c related options : TODO

--- a/CMake/MakefileBuildOptions.cmake
+++ b/CMake/MakefileBuildOptions.cmake
@@ -72,7 +72,7 @@ endforeach()
 # PGI compiler adds --c++14;-A option for C++14, remove ";"
 string(REPLACE ";" " " CXX14_STD_FLAGS "${CMAKE_CXX14_STANDARD_COMPILE_OPTION}")
 string(TOUPPER "${CMAKE_BUILD_TYPE}" _BUILD_TYPE)
-set(CORENRN_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${CMAKE_CXX_FLAGS_${_BUILD_TYPE}} ${CXX14_STD_FLAGS}")
+set(CORENRN_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${CMAKE_CXX_FLAGS_${_BUILD_TYPE}} ${CXX14_STD_FLAGS} ${NVHPC_CXX_INLINE_FLAGS}")
 
 # =============================================================================
 # nmodl/mod2c related options : TODO

--- a/CMake/OpenAccHelper.cmake
+++ b/CMake/OpenAccHelper.cmake
@@ -68,10 +68,10 @@ if(CORENRN_ENABLE_GPU)
   set(CMAKE_CXX14_STANDARD_COMPILE_OPTION --c++14)
   string(APPEND CMAKE_CXX_FLAGS " ${NVHPC_ACC_COMP_FLAGS} ${PGI_DIAG_FLAGS}")
   string(APPEND CMAKE_EXE_LINKER_FLAGS " ${NVHPC_ACC_LINK_FLAGS}")
-  # Use `-Mautoinline` option to compile .cpp files generated from .mod files only.
-  # This is especially needed when we compile with -O0 or -O1 optimisation
-  # level where we get link errors. Use of `-Mautoinline` ensure that the necessary
-  # functions like `net_receive_kernel` are inlined for OpenACC code generation.
+  # Use `-Mautoinline` option to compile .cpp files generated from .mod files only. This is
+  # especially needed when we compile with -O0 or -O1 optimisation level where we get link errors.
+  # Use of `-Mautoinline` ensure that the necessary functions like `net_receive_kernel` are inlined
+  # for OpenACC code generation.
   set(NVHPC_CXX_INLINE_FLAGS "-Mautoinline")
 endif()
 

--- a/CMake/OpenAccHelper.cmake
+++ b/CMake/OpenAccHelper.cmake
@@ -68,6 +68,11 @@ if(CORENRN_ENABLE_GPU)
   set(CMAKE_CXX14_STANDARD_COMPILE_OPTION --c++14)
   string(APPEND CMAKE_CXX_FLAGS " ${NVHPC_ACC_COMP_FLAGS} ${PGI_DIAG_FLAGS}")
   string(APPEND CMAKE_EXE_LINKER_FLAGS " ${NVHPC_ACC_LINK_FLAGS}")
+  # Use `-Mautoinline` option to compile .cpp files generated from .mod files only.
+  # This is especially needed when we compile with -O0 or -O1 optimisation
+  # level where we get link errors. Use of `-Mautoinline` ensure that the necessary
+  # functions like `net_receive_kernel` are inlined for OpenACC code generation.
+  set(NVHPC_CXX_INLINE_FLAGS "-Mautoinline")
 endif()
 
 # =============================================================================

